lib/internal/http2/core.js

```diff
@@ -1745,6 +1728,31 @@ class Http2Stream extends Duplex {
     priorityFn();
   }
 
+  trailers(headers) {
```

Comment:
what will happen if `trailers(headers)` is called _before_ `end()`? This case should be tested.

Reply:
This is very dangerous. I would have `trailers` throw if the event has not be emitted yet instead.